### DISCUSSION TITLE
Screen Mirror - suppress/disable confirmation?

### DISCUSSION THREAD

**Discussion by beer-matt** [BimmerGestalt/AAIdrive#836] [Ideas]
Is it possible to suppress the confirmation when opening screen mirroring (possibly optional disable/suppress)?

I've granted permission via adb (adb shell appops set io.bimmergestalt.idriveconnectaddons.screenmirror PROJECT_MEDIA allow) so that android doesn't prompt about casting/recording every time; would be good to be able to start without needing to touch the phone at all..

**Comment by hufman** [comment]
My understanding of the [VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR flag](https://developer.android.com/reference/android/hardware/display/DisplayManager#VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR) is that it requires a [MediaProjection](https://github.com/BimmerGestalt/IDriveConnectAddons/blob/20692d345a4df72fb0736b8f186a06744b6ca018/screen_mirror/src/main/java/io/bimmergestalt/idriveconnectaddons/screenmirror/RequestActivity.kt#L29) from that system prompt, or some specific system permissions such as `CAPTURE_VIDEO_OUTPUT`. I'm not sure how to [start mirroring](https://github.com/BimmerGestalt/IDriveConnectAddons/blob/20692d345a4df72fb0736b8f186a06744b6ca018/screen_mirror/src/main/java/io/bimmergestalt/idriveconnectaddons/screenmirror/ScreenMirrorProvider.kt#L106-L112) the screen otherwise.
